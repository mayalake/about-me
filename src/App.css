.App {
  font-family: palatino;
}

.Header{
	text-align: center;
	color: white;
	background: linear-gradient(rgba(0,0,0,.3), rgba(0,0,0,.3)), url(./img/background.jpg);
	background-position: center;
	background-repeat: no-repeat;
	background-size: cover;
	height: 100vh;
}

.HeaderText{
	position: absolute;
	top: 50%;
	left: 50%;
	transform: translate(-50%, -50%);
}

.Footer{
	text-align: center;
	color: grey;
	font-size: 12px;
}
