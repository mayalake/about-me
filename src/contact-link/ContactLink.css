.ContactLink{
	margin: 10px 0;
}

.ContactLink a{
	text-decoration: none;
	color: grey;
}
.ContactLink a:hover{
	text-decoration: underline;
}

.IconContainer{
	display: inline-block;
	vertical-align: middle;
	width: 20px;
	margin: 0 10px 0 0;
}

.LinkContainer{
	display: inline-block;
	vertical-align: middle;
	width: 300px;
}

.isFirstLink{
	margin-top: 0;
}