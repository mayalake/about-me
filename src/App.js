import React from 'react';

import NavBar from './navbar/NavBar';
import AboutMeSection from './about-section/AboutMeSection.jsx';
import ResumeSection from './resume-section/ResumeSection';
import ContactSection from './contact-section/ContactSection';

import './App.css';

const App = () =>
    <div className="App">
    	<header className="Header">
			<NavBar />
			<div className="HeaderText">
				<h1>Maya Lake</h1>
				<p>Software Developer<br/>Los Angeles</p>
			</div>
		</header>
		<AboutMeSection />
		<ResumeSection />
		<ContactSection />
		<footer className="Footer">&#169; 2020 Maya Lake</footer>
    </div>;

export default App;
