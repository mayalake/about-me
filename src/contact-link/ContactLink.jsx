import React from 'react';
import PropTypes from 'prop-types';

import './ContactLink.css';

const ContactLink = (props) =>
<div className={`ContactLink ${props.isFirstLink && "isFirstLink"}`}>
	<img className="IconContainer" src={props.image} alt=""/>
	<a className="LinkContainer" href={props.link} target="_blank" rel="noopener noreferrer">{props.text}</a>
</div>;

ContactLink.propTypes = {
	image: PropTypes.string.isRequired,
	text: PropTypes.string.isRequired,
	link: PropTypes.string.isRequired,
	isFirstLink: PropTypes.bool
}

export default ContactLink;