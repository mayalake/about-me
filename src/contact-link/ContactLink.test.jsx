import React from 'react';
import { render } from '@testing-library/react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import App from '../App';
import ContactLink from './ContactLink';

Enzyme.configure({ adapter: new Adapter() });

const contactLinkDetail = {image:"image", text:"text", link:"link"};
let wrapped = Enzyme.shallow(<ContactLink image={contactLinkDetail.image} text={contactLinkDetail.text} link={contactLinkDetail.link}/>);

describe('ContactLink', () => {
	it('renders the ContactLink correctly', () =>{
		expect(wrapped).toMatchSnapshot();
	});
});