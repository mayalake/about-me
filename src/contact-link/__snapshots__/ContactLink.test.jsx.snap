// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactLink renders the ContactLink correctly 1`] = `ShallowWrapper {}`;
